consensus/istanbul/config.go

```diff
@@ -16,29 +16,99 @@
 
 package istanbul
 
-import "math/big"
+import (
+	"math/big"
 
-type ProposerPolicy uint64
+	"github.com/naoina/toml"
+)
+
+type ProposerPolicyId uint64
 
 const (
-	RoundRobin ProposerPolicy = iota
+	RoundRobin ProposerPolicyId = iota
 	Sticky
 )
 
+// ProposerPolicy represents the Validator Proposer Policy
+type ProposerPolicy struct {
+	Id       ProposerPolicyId    // Could be RoundRobin or Sticky
+	Registry []ValidatorSet      // Holds the ValidatorSet for a given block height
+	By       validatorSortByFunc // func that defines how the ValidatorSet should be sorted
```

Comment:
This `By validatorSortByFunc` looks to be quite error prone as it can be set anywhere in the code and second it assumes that an engineer knows some implementation details such as the difference of sorting between `qbft` and `ibft`

Instead, could we use an unexported `type string` field which value can be either `qbft` or `ibft`. Then `ProposerPolicy` exposes a method `SortValidators` that uses the underlying correct `validatorSortByFunc` depending on the `type`